common/app/services/ConfigAgentTrait.scala

```diff
@@ -135,8 +130,12 @@ trait ConfigAgentTrait extends ExecutionContexts with Logging {
   def isFrontHidden(id: String): Boolean =
     configAgent.get().exists(_.fronts.get(id).flatMap(_.isHidden).exists(identity))
 
+  def isEmailFront(id: String): Boolean =
+    getFrontPriorityFromConfig(id).contains(EmailPriority)
+
+  // email fronts are only served if the email-friendly format has been specified in the request
   def shouldServeFront(id: String, isEmailRequest: Boolean = false)(implicit context: ApplicationContext) = getPathIds.contains(id) &&
-    (context.isPreview || (Switches.DisplayHiddenFrontsAsEmails.isSwitchedOn && isEmailRequest) || !isFrontHidden(id))
+    (context.isPreview || !isFrontHidden(id)) && (isEmailRequest || !isEmailFront(id))
```

Comment:
Given the logic is far from being obvious, I still think testing this method would be quite valuable